avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```diff
@@ -0,0 +1,1030 @@
+  virtualNetworkName: string
+
+  @description('Required. The address space of the created virtual network.')
+  addressSpace: string
+
+  @description('Optional. The existing NAT Gateway resource Id. This should be provided if an existing NAT gateway is available to be used. If this parameter is not provided, a new NAT gateway will be created.')
+  natGatewayResourceId: string?
+
+  @description('Optional. The existing public IP address to assosciate with the NAT gateway. This should be provided if an existing public Ip address is available to be used. If this parameter is not provided, a new Public Ip address will be created.')
+  natGatewayPublicIpAddressResourceId: string?
+
+  @description('Optional. The container instance subnet name in the created virtual network. If not provided, a default name will be used.')
+  containerInstanceSubnetName: string?
+
+  @description('Optional. The container instance subnet CIDR in the created virtual network. If not provided, a default subnet prefix will be used.')
+  containerInstanceSubnetAddressPrefix: string?
+
+  @description('Optional. The container app subnet name in the created virtual network. If not provided, a default name will be used.')
+  containerAppSubnetName: string?
+
+  @description('Optional. The container app subnet CIDR in the created virtual network. If not provided, a default subnet prefix will be used.')
+  containerAppSubnetAddressPrefix: string?
+
+  @description('Optional. The container instance subnet address prefix in the created virtual network. If not provided, a default subnet prefix will be used.')
+  containerInstancesubnetPrefix: string?
+
+  @description('Optional. The container registry subnet address prefix in the created virtual network. If not provided, a default subnet prefix will be used.')
+  containerRegistrySubnetPrefix: string?
+
+  @description('Optional. The container registry private DNS zone Id. If not provided, a new private DNS zone will be created.')
+  containerRegistryPrivateDnsZoneId: string?
+
+  @description('Optional. The subnet name for the container registry private endpoint. If not provided, a default name will be used.')
+  containerRegistryPrivateEndpointSubnetName: string?
+
+  @description('Optional. The subnet name for the container app deployment script. Only required if private networking is used. If not provided, a default name will be used.')
+  containerAppDeploymentScriptSubnetName: string?
+
+  @description('Optional. The subnet address prefix for the container app deployment script which is used to start the placeholder Azure DevOps agent. Only required if private networking is used. If not provided, a default subnet prefix will be used.')
+  containerAppDeploymentScriptSubnetPrefix: string?
+
+  @description('Optional. The deployment script private DNS zone Id. If not provided, a new private DNS zone will be created. Only required if private networking is used.')
+  deploymentScriptPrivateDnsZoneId: string?
```

Comment:
If this is a resource ID, I'd recommend to add that detail to the name :)

Reply:
Same comment for other parameters like ` containerRegistryPrivateDnsZoneId`

Reply:
Resolved